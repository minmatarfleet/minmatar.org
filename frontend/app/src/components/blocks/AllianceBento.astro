---
import { getLangFromUrl, useTranslations, useTranslatedPath } from '@i18n/utils';

const lang = getLangFromUrl(Astro.url);
const t = useTranslations(lang);
const translatePath = useTranslatedPath(lang);

interface Props {
    user_is_officer?:   boolean;
    user_is_director?:  boolean;
}

const {
    user_is_officer = false,
    user_is_director = false,
} = Astro.props

import { marked } from 'marked';
import { renderer } from '@helpers/marked';

import Flexblock from '@components/compositions/Flexblock.astro';
import FlexInline from '@components/compositions/FlexInline.astro';

import Button from '@components/blocks/Button.astro';
import ComponentBlock from '@components/blocks/ComponentBlock.astro';

import AugmentationEvEIcon from '@components/icons/AugmentationEvEIcon.astro';
import GroupsEvEIcon from '@components/icons/GroupsEvEIcon.astro';
import CorporationEvEIcon from '@components/icons/CorporationEvEIcon.astro';
---

<FlexInline
    class="[ alliance-reel ]"
    padding_block='0'
    padding_inline='var(--space-xl)'
    max_width='100%'
>
    <ComponentBlock max_width='700px'>
        <Flexblock>
            <FlexInline gap='var(--space-2xs)'>
                <CorporationEvEIcon />
                <h2>{t('corporations')}</h2>
            </FlexInline>
            
            <Flexblock gap='var(--space-m)'>
                <p>{t('corporations.leading_text')}</p>

                <FlexInline>
                    <Button size='sm' href={translatePath('/alliance/corporations/list/')}>
                        {t('view_corporations')}
                    </Button>
                    <Button size='sm' href={translatePath('/alliance/corporations/associates/')}>
                        {t('view_associates_corporations')}
                    </Button>
                </FlexInline>
            </Flexblock>
        </Flexblock>
    </ComponentBlock>

    <ComponentBlock max_width='700px'>
        <Flexblock>
            <FlexInline gap='var(--space-2xs)'>
                <GroupsEvEIcon />
                <h2>{t('groups.page_title')}</h2>
            </FlexInline>
            
            <Flexblock gap='var(--space-m)'>
                <p set:html={marked.parseInline(t('groups.leading_text_p2'), { renderer })} />

                <FlexInline>
                    <Button size='sm' href={translatePath('/alliance/sigs/list/')}>
                        {t('view_groups')}
                    </Button>
                    
                    {user_is_officer &&
                        <Button size='sm' href={translatePath('/alliance/sigs/requests/')}>
                            {t('group_requests')}
                        </Button>
                        
                        <Button size='sm' href={translatePath('/alliance/sigs/members/')}>
                            {t('view_members')}
                        </Button>
                    }
                </FlexInline>
            </Flexblock>
        </Flexblock>
    </ComponentBlock>

    <ComponentBlock max_width='700px'>
        <Flexblock>
            <FlexInline gap='var(--space-2xs)'>
                <AugmentationEvEIcon />
                <h2>{t('teams.page_title')}</h2>
            </FlexInline>
            
            <Flexblock gap='var(--space-m)'>
                <p>{t('teams.leading_text')}</p>

                <FlexInline>
                    <Button size='sm' href={translatePath('/alliance/teams/list/')}>
                        {t('view_teams')}
                    </Button>
                    
                    {user_is_director &&
                        <Button size='sm' href={translatePath('/alliance/teams/requests/')}>
                            {t('team_requests')}
                        </Button>
                        
                        <Button size='sm' href={translatePath('/alliance/teams/members/')}>
                            {t('view_members')}
                        </Button>
                    }
                </FlexInline>
            </Flexblock>
        </Flexblock>
    </ComponentBlock>
</FlexInline>