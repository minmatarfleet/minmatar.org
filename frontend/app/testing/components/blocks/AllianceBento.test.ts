import { experimental_AstroContainer as AstroContainer } from "astro/container";
import { expect, test } from "vitest";
import AllianceBento from "@components/blocks/AllianceBento.astro";

test("AllianceBento defaults", async () => {
  const container = await AstroContainer.create();
  const result = await container.renderToString(AllianceBento, {});

  expect(result).toMatchSnapshot();
});
