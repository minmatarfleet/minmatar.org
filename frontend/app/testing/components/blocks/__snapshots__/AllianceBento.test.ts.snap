// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AllianceBento defaults 1`] = `"<div class="flex-inline [ alliance-reel ]" padding_block="0" padding_inline="var(--space-xl)" max_width="100%" data-astro-cid-ud5fqf5k style="--flex_inline_gap: var(--space-s);--flex_inline_justification: flex-start;">  <div class="wrapper [ card component ] light-transparency" data-mod data-astro-cid-yv35cmqx style="--wrapper_padding_inline: var(--space-l-xl);--wrapper_padding_inline_end: var(--space-l-xl);--wrapper_padding_block: var(--component-padding-block);--wrapper_max_width: 700px;">   <div class="flex-block gap---space-s" data-mod style="--flex_block_gap: var(--space-s);">  <div class="flex-inline" data-astro-cid-ud5fqf5k style="--flex_inline_gap: var(--space-2xs);--flex_inline_justification: flex-start;">  <img loading="lazy" src="/images/icons/48px-CorporationIcon.png" width="48" height="48" alt="EVE Online Corporation Icon"> <h2>Corporations</h2>  </div>   <div class="flex-block gap---space-m" data-mod style="--flex_block_gap: var(--space-m);">  <p>Strong corporations build strong alliances. Learn more about the corporations, CEOs, and membership that make up the Minmatar Fleet Alliance. Perhaps you can submit an application of your own!</p> <div class="flex-inline" data-astro-cid-ud5fqf5k style="--flex_inline_gap: var(--space-s);--flex_inline_justification: flex-start;">  <a href="/alliance/corporations/list/" data-size="sm" style="--button_background_color: var(--fleet-red); --button_color: var(--fleet-yellow)" data-astro-cid-r5e2wo56="true" class="[ button ]"> <div class="flex-inline" data-astro-cid-r5e2wo56="true" data-astro-cid-ud5fqf5k style="--flex_inline_gap: var(--space-xs);--flex_inline_justification: center;">   View  </div>   </a>  <a href="/alliance/corporations/associates/" data-size="sm" style="--button_background_color: var(--fleet-red); --button_color: var(--fleet-yellow)" data-astro-cid-r5e2wo56="true" class="[ button ]"> <div class="flex-inline" data-astro-cid-r5e2wo56="true" data-astro-cid-ud5fqf5k style="--flex_inline_gap: var(--space-xs);--flex_inline_justification: center;">   Associates  </div>   </a>   </div>    </div>   </div>    </div>   <div class="wrapper [ card component ] light-transparency" data-mod data-astro-cid-yv35cmqx style="--wrapper_padding_inline: var(--space-l-xl);--wrapper_padding_inline_end: var(--space-l-xl);--wrapper_padding_block: var(--component-padding-block);--wrapper_max_width: 700px;">   <div class="flex-block gap---space-s" data-mod style="--flex_block_gap: var(--space-s);">  <div class="flex-inline" data-astro-cid-ud5fqf5k style="--flex_inline_gap: var(--space-2xs);--flex_inline_justification: flex-start;">  <img loading="lazy" src="/images/icons/48px-Grouplist.png" width="48" height="48" alt="EVE Online Group List Icon"> <h2>Special Groups of Interest</h2>  </div>   <div class="flex-block gap---space-m" data-mod style="--flex_block_gap: var(--space-m);">  <p>The purpose of SIGs is to maintain closeness in a massive sea of pilots, as described in <a target="_blank" rel="noreferrer noopener nofollow" href="https://docs.google.com/presentation/d/12W-SQUbKAVssdTkM9ixTm_uTkJiDKCXdF6SA2O0B1PQ/edit?usp=sharing">Town Hall #6</a>. Ultimately, we need to expand and create ways to maintain a close-knit feel in a large organization.</p> <div class="flex-inline" data-astro-cid-ud5fqf5k style="--flex_inline_gap: var(--space-s);--flex_inline_justification: flex-start;">  <a href="/alliance/sigs/list/" data-size="sm" style="--button_background_color: var(--fleet-red); --button_color: var(--fleet-yellow)" data-astro-cid-r5e2wo56="true" class="[ button ]"> <div class="flex-inline" data-astro-cid-r5e2wo56="true" data-astro-cid-ud5fqf5k style="--flex_inline_gap: var(--space-xs);--flex_inline_justification: center;">   View  </div>   </a>   </div>    </div>   </div>    </div>   <div class="wrapper [ card component ] light-transparency" data-mod data-astro-cid-yv35cmqx style="--wrapper_padding_inline: var(--space-l-xl);--wrapper_padding_inline_end: var(--space-l-xl);--wrapper_padding_block: var(--component-padding-block);--wrapper_max_width: 700px;">   <div class="flex-block gap---space-s" data-mod style="--flex_block_gap: var(--space-s);">  <div class="flex-inline" data-astro-cid-ud5fqf5k style="--flex_inline_gap: var(--space-2xs);--flex_inline_justification: flex-start;">  <img loading="lazy" src="/images/icons/48px-Augmentations.png" width="48" height="48" alt="EVE Online Augmentation Icon"> <h2>Teams</h2>  </div>   <div class="flex-block gap---space-m" data-mod style="--flex_block_gap: var(--space-m);">  <p>Teams are similar to special interest groups, but without them our alliance would cease to exist. These are people running things behind the curtain to ensure that weâ€™re not going to failcascade in 6 months.</p> <div class="flex-inline" data-astro-cid-ud5fqf5k style="--flex_inline_gap: var(--space-s);--flex_inline_justification: flex-start;">  <a href="/alliance/teams/list/" data-size="sm" style="--button_background_color: var(--fleet-red); --button_color: var(--fleet-yellow)" data-astro-cid-r5e2wo56="true" class="[ button ]"> <div class="flex-inline" data-astro-cid-r5e2wo56="true" data-astro-cid-ud5fqf5k style="--flex_inline_gap: var(--space-xs);--flex_inline_justification: center;">   View  </div>   </a>   </div>    </div>   </div>    </div>    </div>  "`;
